xbmc/pvr/guilib/PVRGUIActionsChannels.cpp

```diff
@@ -77,6 +77,32 @@ void CPVRChannelSwitchingInputHandler::OnInputDone()
     SwitchToChannel(channelNumber);
 }
 
+namespace
+{
+void UpdateActiveGroup(const std::shared_ptr<CPVRChannelGroupMember>& newChannel)
+{
+  const std::shared_ptr<CPVRPlaybackState> playbackState{
+      CServiceBroker::GetPVRManager().PlaybackState()};
+  const std::shared_ptr<CPVRChannelGroupsContainer> groups{
+      CServiceBroker::GetPVRManager().ChannelGroups()};
+  const std::shared_ptr<CPVRChannelGroup> group{
+      groups->Get(newChannel->IsRadio())->GetById(newChannel->GroupID())};
+
+  // Switch group if new channel is not in the active group.
+  if (group && group != playbackState->GetActiveChannelGroup(newChannel->IsRadio()))
```

Comment:
Consider dropping the isRadio check and move it inside `GetActiveChannelGroup(const std::shared_ptr<CPVRChannelGroupMember>&)`. There's no need to give that info in advance, API consumers just want to know the active channel group regardless of the type.

Reply:
Also `Active` is misleading, you're trying to get the channel group of the future channel to later switch to it, so definitely not active.

Reply:
I need to get the currently active group to be able to compare with the desired group. All good as it is.

Reply:
And to get the right type of group (radio or TV) I use the radio flag of the new channel. Welcome to PVR code base. :-)

Reply:
> And to get the right type of group (radio or TV) I use the radio flag of the new channel. Welcome to PVR code base. :-)

Ok misread the line. Anyway, the bool argument could well go away :)